### PR TITLE
Remove che-starter related stuff

### PR DESCRIPTION
**Pull request by Katka92**
### What does this PR do?
Removing all stuff related to che starter and end-to-end tests. 

EDIT: Remove some unused tests as a part of a cleanup. 

### How have you tested this PR?
Running actual suite locally against production.

### CODE REVIEW

**Comment by Katka92** [comment]
@rhopp @ScrewTSW Even if the PR check succeeded, I would appreciate if at least one of you can have a look. Thank you

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianAbstractTestClass.java [hunk 1]
```diff
@@ -1,150 +0,0 @@
...
-import org.openqa.selenium.TimeoutException;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.testng.SkipException;
-import org.testng.annotations.AfterClass;
-import org.testng.annotations.AfterMethod;
-import org.testng.annotations.BeforeMethod;
-import org.testng.annotations.Test;
-
-public abstract class BayesianAbstractTestClass extends RhCheAbstractTestClass {
```
Comment (ibuziuk) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianAbstractTestClass.java]: is this removal che-starter related?

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPackageJson.java [hunk 2]
```diff
@@ -1,48 +0,0 @@
...
- *   Red Hat, Inc. - initial API and implementation
- */
-package com.redhat.che.functional.tests;
-
-import com.redhat.che.selenium.core.workspace.RhCheWorkspaceTemplate;
-import org.eclipse.che.selenium.core.workspace.InjectTestWorkspace;
-import org.eclipse.che.selenium.core.workspace.TestWorkspace;
-import org.testng.annotations.BeforeClass;
-
-public class BayesianPackageJson extends BayesianAbstractTestClass {
```
Comment (ibuziuk) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPackageJson.java]: is this removal che-starter related?

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPomXml.java [hunk 3]
```diff
@@ -1,85 +0,0 @@
...
-import com.google.inject.Inject;
-import com.redhat.che.selenium.core.workspace.RhCheWorkspaceTemplate;
-import org.eclipse.che.selenium.core.workspace.InjectTestWorkspace;
-import org.eclipse.che.selenium.core.workspace.TestWorkspace;
-import org.eclipse.che.selenium.pageobject.CodenvyEditor;
-import org.openqa.selenium.TimeoutException;
-import org.testng.SkipException;
-import org.testng.annotations.BeforeClass;
-
-public class BayesianPomXml extends BayesianAbstractTestClass {
```
Comment (ibuziuk) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPomXml.java]: is this removal che-starter related?

File: functional-tests/src/test/java/com/redhat/che/functional/tests/EETest.java [hunk 4]
```diff
@@ -1,101 +0,0 @@
...
-import org.eclipse.che.selenium.pageobject.Loader;
-import org.eclipse.che.selenium.pageobject.Menu;
-import org.eclipse.che.selenium.pageobject.ProjectExplorer;
-import org.eclipse.che.selenium.pageobject.git.Git;
-import org.eclipse.che.selenium.pageobject.git.GitCompare;
-import org.testng.annotations.BeforeClass;
-import org.testng.annotations.Test;
-
-// class for trying ProvidedWorkspace functionality
-public class EETest extends RhCheAbstractTestClass {
```
Comment (ibuziuk) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/EETest.java]: good to remove this test - naming is really poor

File: functional-tests/src/test/java/com/redhat/che/functional/tests/TestTestClass.java [hunk 5]
```diff
@@ -1,88 +0,0 @@
...
-
-import com.google.inject.Inject;
-import com.redhat.che.selenium.core.workspace.CheStarterWrapper;
-import com.redhat.che.selenium.core.workspace.RhCheTestWorkspaceImpl;
-import org.eclipse.che.selenium.core.user.DefaultTestUser;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.testng.annotations.Test;
-
-public class TestTestClass {
```
Comment (ibuziuk) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/TestTestClass.java]: TestTestClass is funny :-)

**Comment by ibuziuk** [comment]
@Katka92 overall looks good, please clarify if all the removal is che-starter related or there are some other cleanup actions also

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianAbstractTestClass.java (hunk 1 above)
Comment (Katka92) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianAbstractTestClass.java]: This test is not used as far as issues with Bayesian are not planned to be fixed. I've added it here as a part of a cleanup. I'll mention it in a PR description.

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPackageJson.java (hunk 2 above)
Comment (Katka92) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPackageJson.java]: This test is not used as far as issues with Bayesian are not planned to be fixed. I've added it here as a part of a cleanup. I'll mention it in a PR description.

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPomXml.java (hunk 3 above)
Comment (Katka92) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPomXml.java]: This test is not used as far as issues with Bayesian are not planned to be fixed. I've added it here as a part of a cleanup. I'll mention it in a PR description.

Comment (ibuziuk) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPomXml.java]: well, but Bayesian should work on prod, right ?

Comment (ibuziuk) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPomXml.java]: I'm huge -1 to remove the testcase, it should be run on prod and ignored on prod-preview

**Comment by Katka92** [comment]
There is an issue created some time ago. https://github.com/redhat-developer/rh-che/issues/1256 Bayesian was skipped on prod-preview and was failing on a production. I understand the situation in a way that you're not planning to fix Bayesian for Che 6 (as it should be deprecated soon), so we decided to remove tests.

**Comment by ibuziuk** [comment]
yeah, we do not plan to work on https://github.com/redhat-developer/rh-che/issues/1256, but this is not the reason to remove test, it could just test initial openning and reopening could be coverred in different test which would be ignored with the link to the issue

**Comment by Katka92** [comment]
So you are planning to maintain Bayesian but only for initial opening of the file, right? So I should enable the test and if that part fails, I should create an issue and you will find some time in a team to fix that. Am I getting it right? So only a feature "showing an error on reopening" will not be supported anymore. Otherwise I see no reason to keep that test.

**Comment by ibuziuk** [comment]
> So you are planning to maintain Bayesian but only for initial opening of the file, right? So I should enable the test and if that part fails, I should create an issue and you will find some time in a team to fix that. Am I getting it right?

Correct, you are getting it right.

>  So only a feature "showing an error on reopening" will not be supported anymore. Otherwise, I see no reason to keep that test.

it will be supported if we would have the capacity to fix it, otherwise it would be known issue reflected in tests

**Comment by Katka92** [comment]
Ok, then I'll enable the test again.

**Comment by ibuziuk** [comment]
just to make it clear - all tests should either pass or being ignored with the link to dedicated issue e.g. case
Redundant tets e.g. che-starter specific should be removed. There should be no flaky tests

**Comment by Katka92** [comment]
I agree that there should not be any flaky tests. It make sense to have some tests skipped if there is a possibility in future that they will get fixed. But having failing test which is not supposed to be fixed is just waste of a time - it will bring no value and make all checks (PR or periodic) just last longer and log more unusable info.

**Comment by ibuziuk** [comment]
> But having failing test which is not supposed to be fixed is just waste of a time 

Why do you say that tests are not supposed to be fixed? we just do not have the capacity to fix this atm

**Comment by Katka92** [comment]
I know you have many fires to extinguish and I read many times that Che 6 is going to be deprecated soon. So that's why I though you are not gonna support that and invest time here. Probably I was getting it wrong, so I've added Bayesian tests to periodic and PR check suites. Che-starter related tests are removed.

**Comment by Katka92** [comment]
[test]

**Comment by Katka92** [comment]
@ibuziuk Can you review again? I've added Bayesian to periodic tests. Can I merge it?

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPomXml.java [hunk 6]
```diff
@@ -44,14 +43,16 @@
 
   @BeforeClass
   public void setUp() throws Exception {
+    if (cheHost.equals(CHE_PROD_PREVIEW_URL)) {
```
Comment (ibuziuk) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPomXml.java]: @Katka92 would be nice to add links to the issues there in order to clarify why the tests are skipped

**Comment by ibuziuk** [comment]
@Katka92 overall LGTM, some minor comment for adding issue ref

**Pull request merged**